internal/policy/container/has_modified_files.go

```diff
@@ -241,7 +261,9 @@ func (p *HasModifiedFilesCheck) validate(ctx context.Context, layerIDs []string,
 
 			if (previousOsRelease && !currentOsRelease) || (previousPackage.Arch != currentPackage.Arch) {
 				// If either of these differ, that's a fail
-				return false, nil
+				logger.V(log.DBG).Info("mismatch in package architecture", "file", modifiedFile)
```

Comment:
Is this phrasing correct? This log line seems to imply only the arch could be wrong here. Should we ever hit this case for x86_64 || i686?

Reply:
We will not hit this case coded for the above code. This was the code block that was originally causing a failure and were never logging this and it caused confusion for cert-ops. I'd like to make this line better, but I need to re-read @bcrochet comments about what this if block covers and comment accordingly, that should yield a better log message.

Reply:
Let's make them more explicit. It makes sense to split these two so that we get a more granular message.
This is still valid. This is the case that someone replaced a Red Hat package with one of a different arch. OR, they did a rebuild, and the release changed. 

EIther way, it shouldn't pass. But making it separate will make the code more readable.

Reply:
I split this into two different `if` conditions and added a unit test to cover both cases.